src/utils/hooks/useGetReviewInfo.tsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect, useState } from 'react'
+import { AssignmentDetailsNEW } from '../types'
+import { Review, ReviewAssignment, ReviewStatus, useGetReviewInfoQuery } from '../generated/graphql'
+import messages from '../messages'
+
+const MAX_REFETCH = 10
+interface UseGetReviewInfoProps {
+  applicationId: number
+  userId: number
+}
+
+const useGetReviewInfo = ({ applicationId, userId }: UseGetReviewInfoProps) => {
+  const [assignments, setAssignments] = useState<AssignmentDetailsNEW[]>()
+  const [isFetching, setIsFetching] = useState(true)
+  const [fetchingError, setFetchingError] = useState('')
+  const [refetchAttempts, setRefetchAttempts] = useState(0)
+
+  const { data, loading, error, refetch } = useGetReviewInfoQuery({
+    variables: {
+      reviewerId: userId,
+      applicationId,
+    },
+    notifyOnNetworkStatusChange: true,
+  })
+
+  useEffect(() => {
+    if (!data || loading) return
+    const reviewAssigments = data.reviewAssignments?.nodes as ReviewAssignment[]
+
+    // Current user has no assignments
+    if (!reviewAssigments) {
+      setIsFetching(false)
+      return
+    }
+
+    const reviews: Review[] = reviewAssigments.map(({ reviews }) => reviews.nodes[0] as Review)
+
+    // Checking if any of reviews trigger is running before refetching assignments
+    // This is done to get latest status for reviews (afte trigger finishes to run)
+    if (reviews.every(({ trigger }) => trigger === null)) {
```

Comment:
` if (reviews.every((review) => !review || review?.trigger === null)) {` (otherwise forever loading if no reviews)

Reply:
Ooh, no! :D Nice catch! Done: a391b36

Reply:
Still not quite right, but almost there ( i think my suggestion works nicely)
* if no reviews but review_assignments, there is an error that trigger is not defined
* if that error is fixed, but we are checking for .length of review_assignments, not review (since mapping over review assignments)

Reply:
8a870d0702f4065cb8faddbf016a946136f6cc34